### ISSUE TITLE
Allow specifying a journal by filename in arguments

### ISSUE THREAD

**Issue by xeruf** [jrnl-org/jrnl#1558]
### Use Case/Motivation

Sometimes I shuffle my journals around and resolve conflicts from different devices, and then it would be helpful to operate on a journal by filename

### Example Usage

Not only try to find a journal in `journals` config by name, but also check if there is a file with that name/path and use that.

    jrnl intentions.txt Be Happy
    
Would insert "Be Happy" into a jrnl file called `intentions.txt` if it exists.

**Comment by micahellison** [comment]
Hi @xeruf, thanks for filing this issue. Currently, this is possible through the `--config-override argument`, with some caveats. If you have a default jrnl configuration, you can journal to a different file like so:
```
jrnl --config-override journals.default ./intentions.txt Be Happy
```

The caveat is that if your config file is structured differently such that there settings specific to your journal (for instance, if you encrypt your journal), you'll have to refer to the journal's path using the "journal" config key of your journal like so:
```
jrnl --config-override journals.default.journal ./intentions.txt Be Happy
```
There's an open bug #1224 out there to simplify that issue, but even after it's resolved, the above methods should still work.

**Comment by wren** [comment]
How about if we use the `--file` flag? So, it might look something like:

```
jrnl --file intentions.txt Be Happy
```

**Comment by xeruf** [comment]
why a flag at all if there is already keyword recognition for specifying journals?

**Comment by wren** [comment]
@xeruf The keyword recognition of journal names works because the config already has to be parsed in order for jrnl to function. So, when a user enter something like `jrnl hello my name is`, it's very quick and easy to check if `hello` is the name of a journal or not, and proceed accordingly.

If we expand this functionality to also include filenames, jrnl would have to check the current folder each time a command is run. This could have various downsides (jrnl performance would go down, commands would have inconsistent results depending on what folder you're in when you run them, etc). And that's not even getting into what jrnl should do if the file doesn't exist (it currently creates it for you, but there's no surefire way to know if a given word is intended to be filename from the user).

Even with the idea of using `--file`, we'll have to figure out how to deal with the various formats that jrnl supports (one file, encrypted, directory journal, etc), since we no longer have the config to give us that information.

So, in summary, the current magic works because of the information we have in the config. If we take it out of the config, we have to have some way to reliably determine the intent of the user.

**Comment by xeruf** [comment]
fair points, though `jrnl` anyways does a file access to write to the journal ^^
maybe this could also be a config option, whether it tries to match to a filename

> And that's not even getting into what jrnl should do if the file doesn't exist (it currently creates it for you, but there's no surefire way to know if a given word is intended to be filename from the user).

That's easy, it should only act differently if the first word is an existing text file.

**Comment by micahellison** [comment]
I'm not a big fan of making this feature it "magical" like journal names currently are. It would mean jrnl would have to check the current directory's files every time it runs, which could be particularly annoying for a user depending on where the current directory is. It's a lot different than the regular config file access that only happens in the user folder.

I also worry about the impact of journaling in a directory with a lot of files. The user would have to be aware of this feature so as to avoid the filename of any file whenever journaling a short entry from the command line.

That said, an explicit argument with hyphens for a prefix would mitigate all these issues. And ideally, that argument could be abbreviated, like `--f` as an alias for `--file`.

**Comment by xeruf** [comment]
yeah I guess a flag is fine
